tokio/src/runtime/io/scheduled_io.rs

```diff
@@ -81,48 +82,31 @@ enum State {
 
 // The `ScheduledIo::readiness` (`AtomicUsize`) is packed full of goodness.
 //
-// | shutdown | generation |  driver tick | readiness |
-// |----------+------------+--------------+-----------|
-// |   1 bit  |   7 bits   +    8 bits    +   16 bits |
+// | shutdown | driver tick | readiness |
+// |----------+-------------+-----------|
+// |   1 bit  |   8 bits    +   16 bits |
 
 const READINESS: bit::Pack = bit::Pack::least_significant(16);
 
 const TICK: bit::Pack = READINESS.then(8);
 
-const GENERATION: bit::Pack = TICK.then(7);
-
-const SHUTDOWN: bit::Pack = GENERATION.then(1);
-
-#[test]
-fn test_generations_assert_same() {
-    assert_eq!(super::driver::GENERATION, GENERATION);
-}
+const SHUTDOWN: bit::Pack = TICK.then(1);
 
 // ===== impl ScheduledIo =====
 
-impl Entry for ScheduledIo {
-    fn reset(&self) {
-        let state = self.readiness.load(Acquire);
-
-        let generation = GENERATION.unpack(state);
-        let next = GENERATION.pack_lossy(generation + 1, 0);
-
-        self.readiness.store(next, Release);
-    }
-}
-
 impl Default for ScheduledIo {
     fn default() -> ScheduledIo {
         ScheduledIo {
+            linked_list_pointers: UnsafeCell::new(linked_list::Pointers::new()),
             readiness: AtomicUsize::new(0),
             waiters: Mutex::new(Default::default()),
         }
     }
 }
 
 impl ScheduledIo {
-    pub(crate) fn generation(&self) -> usize {
-        GENERATION.unpack(self.readiness.load(Acquire))
+    pub(crate) fn token(&self) -> mio::Token {
+        mio::Token(self as *const _ as usize)
```

Comment:
similarly to my above comment, this is where we would want to use `pointer::expose_addr` eventually, if it becomes stable..